src/desktop/space/mod.rs

```diff
@@ -668,3 +670,54 @@ where
 
     damage_tracked_renderer.render_output(renderer, age, &*render_elements, clear_color, log)
 }
+
+/// Find the primary output for a [`WlSurface`] of a window out of a set of outputs
+///
+/// Returns `None` if no primary output could be determined
```

Comment:
Maybe expand on this. In which cases can this happen? If the output is not "visible" on any output?

What visible means in this context is also questionable. Again using the output state here is quite clever, because it works independently from using either the `Space`-abstraction or driving `output_enter` and `output_leave` manually. But for downstream this is not apparent.

Reply:
Updated the docs to include the reason why `None` would be returned and added the requirement of `output_enter`/`output_leave`